packages/bot/src/bot.ts

```diff
@@ -101,11 +104,15 @@ export interface CreateBotOptions {
   /** The bot's intents that will be used to make a connection with discords gateway. */
   intents?: GatewayIntents
   /** Any options you wish to provide to the rest manager. */
-  rest?: CreateRestManagerOptions
+  rest?: CreateRestManagerOptions & Partial<Pick<CreateRestManagerOptions, 'token'>>
   /** Any options you wish to provide to the gateway manager. */
-  gateway?: CreateGatewayManagerOptions
+  gateway?: CreateGatewayManagerOptions & Partial<Pick<CreateGatewayManagerOptions, 'token'>>
   /** The event handlers. */
-  events: Partial<EventHandlers>
+  events?: Partial<EventHandlers>
+  /** The functions that should transform discord objects to discordeno shaped objects. */
+  transformers?: Partial<Transformers>
+  /** The handler functions that should handle incoming discord payloads from gateway and call an event. */
+  handlers?: Partial<BotGatewayHandlerOptions>
```

Comment:
```suggestion
  /** The functions that should customize the discordeno shaped objects */
  customizers?: Partial<Transformers["customizers"]>
```

I usually think of transformers and handlers of a more internal part of discordeno rather then something that should be exposed, especially in the createBot, for example the handlers 99% of the time you are not going to edit them because if you do it's not something common so overring the function after the creation is not that big of deal, same for transformers, they tend to be code that you don't want to mained as they require a lot of if checks for both proprieties that need to exist and desired proprieties that need to be enabled or even worse if the transformer has a base template (eg. Message, Interaction ecc) as those would be impossibile to change in here without coping that entire part of the code

Reply:
For handlers, I could agree that maybe those are "internal enough" to not be exposed, but I feel like there are still very valid cases for overriding/plugging into transformers, the immediately obvious use for it being caching.

On the other hand, I believe that even if these are/could be considered internals, whether that's transformers or events or handlers, allowing the developer to override them in a consistent interface if they need to is still helpful. After all, Discordeno was designed for big bot developers who are more or less expected to know the outcomes of what they plug into or change in the library.

Reply:
> the immediately obvious use for it being caching.

I do not see the value of caching the parsing of structure at all, caching and merging the objects can be done in the customizers


> After all, Discordeno was designed for big bot developers who are more or less expected to know the outcomes of what they plug into or change in the library.

You do have a point, although i still personally still feel they don't belong, so for now i will leave the conversation as "unresolved" to see if anyone else as an opinion on this

Reply:
After hearing both sides, I prefer to lean towards the side that DD should be flexible and built for those who know how to work with it.